scripts/all-tests.sh

```diff
@@ -1,34 +1,48 @@
 #!/usr/bin/env bash
 set -euo pipefail
 
+log() {
+  echo "[$(date +'%Y-%m-%dT%H:%M:%S%z')] $*"
+}
+
 # Ensure dependencies are tidy and up to date
-echo "Tidying go modules and checking for changes..."
+log "Tidying Go modules and checking for changes..."
 go mod tidy
-git diff --exit-code go.mod go.sum || {
-  echo "go.mod or go.sum has uncommitted changes after running 'go mod tidy'."
+if ! git diff --exit-code go.mod go.sum; then
+  log "go.mod or go.sum has uncommitted changes after 'go mod tidy'."
   exit 1
-}
+fi
 
-# Ensure go code is formatted properly
-echo "Checking code format with 'go fmt'..."
-# Capture the result of go fmt into a variable
+# Ensure Go code is formatted properly
+log "Checking code format with 'go fmt'..."
 fmt_output=$(go fmt ./...)
-
 if [ -n "$fmt_output" ]; then
-  echo "The following files are not formatted properly:"
+  log "The following files are not formatted properly:"
   echo "$fmt_output"
-  echo "Please run 'go fmt' and fix the formatting issues before committing."
+  log "Please run 'go fmt' to fix the formatting issues."
   exit 1
 fi
 
 # Run golangci-lint to check for code issues
-echo "Running golangci-lint..."
-golangci-lint run
+log "Running golangci-lint..."
+if ! golangci-lint run; then
+  log "Linting issues detected."
+  exit 1
+fi
 
+# Run tests in parallel for faster execution
+log "Running unit tests..."
 cd "$( dirname "${BASH_SOURCE[0]}" )/.."
-./scripts/unit.sh
-echo
-./scripts/integration.sh
-echo
-./scripts/contract.sh
+./scripts/unit.sh &
```

Comment:
**issue:** I tested this out just now (note that it is still early :)): 
```
./scripts/unit.sh & unit_pid=$!; ./scripts/integration.sh & integration_pid=$!; wait $unit_pid $integration_pid
```
I made my unit test fail, but the integration tests still ran and the exit code was 0:
```
± |main {6} U:1 ✗| → echo $?
0
```
We would want to fail with a non zero exit code if any of the steps fail (including the test). I think we could skip all further steps if one step fails if that makes sense. Ie. don't run integration tests when unit tests fail. Don't run unit tests when the linter fails etc. We could print a message in which we explain that steps get skipped though? What do you think?

Reply:
Nice thought @kardolus

Reply:
@kardolus Sir will you  merge it?

Reply:
@Innovatorcloudy I can merge it in and then make a few changes myself. Or you can do them if you'd prefer. Either way works for me.